python/pyarrow/tests/test_feather.py

```diff
@@ -805,6 +805,16 @@ def test_read_column_duplicated_in_file(tempdir):
         read_table(path, columns=['a', 'b'])
 
 
+def test_read_column_wrong_column_type(tempdir, version):
+    # duplicated columns in the column selection
+    table = pa.table([[1, 2, 3], [4, 5, 6], [7, 8, 9]], names=['a', 'b', 'c'])
+    path = str(tempdir / "data.feather")
+    write_feather(table, path, version=version)
+    columns_gen = (x for x in ['a', 'b', 'c'])
+    with pytest.raises(TypeError):
```

Comment:
Could we match the message also?

```suggestion
    with pytest.raises(TypeError, match="Columns must be a sequence"):
```